src/Extension/FluentExtension.php

```diff
@@ -835,4 +846,118 @@ public function cacheKeyComponent()
+                }
+
+                // Highlight any translatable field
+                if ($field && !$field->hasClass('LocalisedField')) {
+                    // Add a language indicator next to the fluent icon
+                    $currentLocale = Locale::getCurrentLocale();
+                    $locale = $currentLocale->Locale;
+                    $language = $currentLocale ? strtok($locale, '_') : '';
+                    $title = $field->Title();
+
+                    $titleClasses = 'fluent-locale-label';
+                    $modifiedTitle = 'Using default locale value';
+
+                    // Add a visual indicator for whether the value has been changed from the default locale
+//                    if (Fluent::isFieldModified($this->owner, $field, $locale)) {
+//                        $titleClasses .= ' fluent-modified-value';
+//                        $modifiedTitle = 'Modified from default locale value - click to reset';
+//                    }
+
+                    $field->setTitle(
+                        DBHTMLText::create()->setValue(
+                        sprintf(
+                            '<span class="%s" title="%s">%s</span>%s',
+                            $titleClasses,
+                            $modifiedTitle,
+                            $language,
+                            $title
+                        ))
+                    );
+
+                    // Set the default value to the element so we can compare it with JavaScript
+//                    if (Locale::getDefault() !== $locale) {
+//                        $field->setAttribute(
+//                            'data-default-locale-value',
+//                            $this->owner->{Fluent::db_field_for_locale($field->getName(), Locale::getDefault())}
+//                        );
+//                    }
+
+                    $field->addExtraClass('LocalisedField');
+                }
+
+                // Remove translation DBField from automatic scaffolded fields
+//                foreach (Fluent::locales() as $locale) {
+//                    $fieldName = Fluent::db_field_for_locale($translatedField, $locale);
+//                    $fields->removeByName($fieldName, true);
+//                }
+            }
+        }
+
+        $this->addLocaleIndicatorMessage($fields);
+    }
+
+    /**
+     * Adds a UI message to indicate whether you're editing in the default locale or not
+     *
+     * @param  FieldList $fields
+     * @return $this
+     */
+    protected function addLocaleIndicatorMessage(FieldList $fields)
+    {
+        if (Config::inst()->get(__CLASS__, 'disable_current_locale_message')) {
+            return $this;
+        }
+
+        // If the field is already present, don't add it a second time
+        if ($fields->fieldByName('CurrentLocaleMessage')) {
+            return $this;
+        }
+
+        $localeNames     = Locale::get()->map('Locale', 'Title')->toArray();
```

Comment:
If you use `Locale::getCached()` you'll get a minor performance benefit, but follow the same API use as the rest of the module